### PR TITLE
drainer: Refactor and add unit tests

### PR DESCRIPTION
**Pull request by suzaku**
### What problem does this PR solve? 

[Jira Issue](https://internal.pingcap.net/jira/browse/TOOL-1044)

### What is changed and how it works?

Test coverage: 46.3% -> 58.3%

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

### CODE REVIEW

File: drainer/collector_test.go [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+
+	item := binlogItem{
+		binlog: &pb.Binlog{DdlJobId: 1024, CommitTs: 1305432},
+	}
+
+	err := col.syncBinlog(&item)
+	c.Assert(err, IsNil)
+
+	c.Assert(item.job.ID, DeepEquals, job.ID)
+	c.Assert(item.job.State, DeepEquals, job.State)
```
Comment (kennytm) [inline drainer/collector_test.go]: Either you mean `c.Assert(item.job, DeepEquals, job)` or you mean to just use `Equals` here.

Comment (suzaku) [reply drainer/collector_test.go]: Guess I tried to use `c.Assert(item.job, DeepEquals, job)` and then found out it was a huge struct and then gave up.

**Comment by suzaku** [comment]
@kennytm PTAL

**Pull request merged**